### PR TITLE
[jjo] add setup.py and snapcraft.yaml

### PR DESCRIPTION
**Pull request by jjo**
* added setup.py
* added snapcraft.yaml and wrapper for easier $SNAPDATA yaml config
- verified with:
```
snapcraft snap
snapcraft install --devmode prometheus-openstack-exporter*.snap
## dummy runs below, will fail from missing actual YAML and creds
snap run prometheus-openstack-exporter
sudo systemctl start snap.prometheus-openstack-exporter.prometheus-openstack-exporter.service
```

### CODE REVIEW

File: snap_cmd_wrapper [hunk 1]
```diff
@@ -0,0 +1,6 @@
+#!/bin/sh
+
+# Copy only if not existing
+: ${SNAP?} ${SNAPDATA?}
```
Comment (jacekn) [inline snap_cmd_wrapper]: What's the purpose of this line? It looks like old debug statement? Also it should say SNAP_DATA not SNAPDATA

File: snap_cmd_wrapper [hunk 2]
```diff
@@ -0,0 +1,6 @@
...
+cp -pn $SNAP/etc/prometheus/prometheus-openstack-exporter.yaml.example $SNAP_DATA/prometheus-openstack-exporter.yaml
+exec $SNAP/bin/prometheus-openstack-exporter $SNAP_DATA/prometheus-openstack-exporter.yaml
```
Comment (jacekn) [inline snap_cmd_wrapper]: If you ship default config but no novarc the daemon will not start. It's not a major problem for software that requires creds to run but we should make a note to handle this gracefully in the future. Good error message in the logs or maybe user friendly text returned after HTTP call should do it

Comment (jacekn) [inline snap_cmd_wrapper]: 2 minor points but not critical. Would be good to fix them when convenient though.

File: snap_cmd_wrapper (hunk 1 above)
Comment (jjo) [inline snap_cmd_wrapper]: the construct
```
: S{VARNAME?}
```
will abort the shell if var doesn't exist - added it to explicitly fail things direct runs of the wrapper as indeed those are really needed, ie kinda environment validation.
+1 ofc on the SNAP_DATA fix

File: snap_cmd_wrapper (hunk 2 above)
Comment (jjo) [inline snap_cmd_wrapper]: +1 as per discussion, we'd merge this as-is and file a bug to address above.

**Pull request merged**